src/shared/varlink-io.systemd.Resolve.c

```diff
@@ -32,6 +32,33 @@ static VARLINK_DEFINE_METHOD(
                 VARLINK_DEFINE_OUTPUT_BY_TYPE(names, ResolvedName, VARLINK_ARRAY),
                 VARLINK_DEFINE_OUTPUT(flags, VARLINK_INT, 0));
 
+static VARLINK_DEFINE_STRUCT_TYPE(
+                ResolvedService,
+                VARLINK_DEFINE_FIELD(priority, VARLINK_INT, 0),
+                VARLINK_DEFINE_FIELD(weight, VARLINK_INT, 0),
+                VARLINK_DEFINE_FIELD(port, VARLINK_INT, 0),
+                VARLINK_DEFINE_FIELD(hostname, VARLINK_STRING, 0));
+
+static VARLINK_DEFINE_STRUCT_TYPE(
+                ResolvedCanonical,
+                VARLINK_DEFINE_FIELD(name, VARLINK_STRING, 0),
+                VARLINK_DEFINE_FIELD(type, VARLINK_STRING, 0),
+                VARLINK_DEFINE_FIELD(domain, VARLINK_STRING, 0));
+
+static VARLINK_DEFINE_METHOD(
+                ResolveService,
+                VARLINK_DEFINE_INPUT(name, VARLINK_STRING, 0),
+                VARLINK_DEFINE_INPUT(type, VARLINK_STRING, 0),
+                VARLINK_DEFINE_INPUT(domain, VARLINK_STRING, 0),
+                VARLINK_DEFINE_INPUT(ifindex, VARLINK_INT, VARLINK_NULLABLE),
+                VARLINK_DEFINE_INPUT(family, VARLINK_INT, VARLINK_NULLABLE),
+                VARLINK_DEFINE_INPUT(flags, VARLINK_INT, VARLINK_NULLABLE),
+                VARLINK_DEFINE_OUTPUT_BY_TYPE(srv, ResolvedService, 0),
+                VARLINK_DEFINE_OUTPUT_BY_TYPE(addr, ResolvedAddress, VARLINK_ARRAY),
+                VARLINK_DEFINE_OUTPUT(txt, VARLINK_STRING, VARLINK_ARRAY),
```

Comment:
Looks like this is triggered when services comes with no key/value pairs:
```
Varlink-IDL: Mandatory field 'txt' is null or missing on object, refusing.
varlink-26: Return parameters for method reply ResolveService() didn't pass validation on field 'txt', ignoring: No anode
varlink-26: Sending message: {"parameters":{"srv":{"priority":0,"weight":0,"port":123,"hostname":"host-5.local"},"addr":[{"ifindex":3,"family":2,"address":[192,168,56,3]},{"ifindex":3,"family":10,"address":[254,128,0,0,0,0,0,0,0,224,25,56,10,209,68,157]}],"txt":null,"normalized":"host-5.local","canonical":{"name":"X","type":"_ipp._tcp","domain":"local"}}}
```
it should probably emit empty arrays instead of `null` when it receives TXT RRs with a single empty string
```
            Name: X._ipp._tcp.local                                                                                                                           
            Type: TXT (Text strings) (16)                                                                                                                     
            .000 0000 0000 0001 = Class: IN (0x0001)                                                                                                          
            1... .... .... .... = Cache flush: True                                                                                                           
            Time to live: 4500 (1 hour, 15 minutes)                                                                                                           
            Data length: 1                                                                                                                                    
            TXT Length: 0                                                                                                                                     
            TXT:                                  
```
or zero-length RDATA.